esp-mbedtls/src/lib.rs

```diff
@@ -244,12 +244,133 @@ impl<'a> X509<'a> {
     }
 }
 
-/// Certificates used for a connection.
-///
-/// # Note:
-/// Both [certificate](Certificates::certificate) and [private_key](Certificates::private_key) must be set in pair.
-#[derive(Debug, Clone, Copy, PartialEq)]
-pub struct Certificates<'a> {
+/// Creates a wrapper over [mbedtls_x509_crt] to safely manage allocation and freeing on drop
+#[derive(Debug)]
+struct MbedTLSX509Crt(*mut mbedtls_x509_crt);
```

Comment:
Changing this to 
```rust
MbedTlsX509Crt<'d> {
    crt: *mut mbedtls_x509_crt,
    _t: PhantomData<&'d ()>,
}
```

... would help express the fact that the `*mut mbedtls_x509_crt` _might_ be keeping a reference to a `X509` cert contents.

}